### PR TITLE
Document mTLS support

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Refs; https://github.com/TheThingsIndustries/lorawan-stack/pull/3397
https://github.com/TheThingsIndustries/lorawan-stack-aws/pull/681

#### Screenshots

Screen Shot 2022-08-30 at 10 05 52

#### Changes

- Add new mTLS Section

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [x] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+But since credentials once stored in ACM cannot be retrieved (can only be referenced by NLB), we need to store a copy of these credentials in the AWS Secrets Manager.
+
+Deploy the `4-1-secrets` template to create a new secret in ACM. This also generates additional keys in the KeyVault for {{% tts %}} components to exchange auth information.
+
+Then deploy `5-7a-certs-le` to update the Certbot task and fetch new certificates using the [instructions]({{< relref "#lets-encrypt-certificates-optional" >}}).
+
+### Step 3: Update the Load Balancer and the Proxy
+
+{{< warning >}}
+This step requires brief downtime. Make sure to inform your customers
```
Comment (adriansmares) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:63]: Nit: dot at the end.

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+We are now modifying the protocol of the NLB ports `443` and `8887` to TCP. This means that the target groups that these ports use must be modified.
+
+Since the Proxy service is already bound to these target groups, CloudFormation would error if you try to update this.
+
+First, make a copy of the configuration parameters in the `5-6-ecs-proxy` CloudFormation stack.
+
+Next, delete this CloudFormation stack.
+
+{{< note >}}
+This will only delete the Proxy ECS containers and alarms. This can be easily reverted by re-deploying `5-6-ecs-proxy`
```
Comment (adriansmares) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:79]: Nit: dot at the end.

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+This is only for testing. Do not use this process in production.
+{{</ warning >}}
+
+1. Create a client gateway private key. Note you can choose any length of the key.
+
+```bash
+$ openssl genrsa -des3 -out client.key 2048
+```
+
+Unlock the key
```
Comment (adriansmares) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:122]: Nit: Colon at the end.

**Review by adriansmares — COMMENTED** [review]
This LGTM modulo some insignificant nits. But let's merge the implementation (TTS+ECS) first.

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+In the standard deployment model, AWS Network Load Balancer (NLB) terminates TLS, and then forwards unencrypted packets to {{% tts %}} via AWS internal network.
+
+Since the NLB does not support verifying client TLS certificates, the responsibility for terminating TLS is promoted to the proxy (only) for the **necessary ports**.
+
+The other ports will have TLS terminated by the NLB.
+
+### Certificate Requirements
+
+The `CN` (Common Name) field of the certificate must be the 16 digit gateway EUI. ex `1111111111111111`
```
Comment (nejraselimovic) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:30]:
```suggestion
The `CN` (Common Name) field of the certificate must be the 16 digit gateway EUI (example: `1111111111111111`).
```

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+2. Create a Root CA
+
+```bash
+$ openssl req -x509 -sha256 -days 1825 -newkey rsa:2048 -keyout root-ca.key -out root-ca.crt
+```
+
+3. Create a Certificate Signing Request for your client (Gateways)
+
+{{< note >}}
+The `CN` (Common Name) field must be the 16 digit gateway EUI. ex `1111111111111111`
```
Comment (nejraselimovic) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:137]:
```suggestion
The `CN` (Common Name) field must be the 16 digit gateway EUI (example: `1111111111111111`).
```

File: doc/content/getting-started/aws/ecs/mutual-tls/_index.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+```bash
+$ openssl x509 -req -CA root-ca.crt -CAkey root-ca.key -in gateway.csr -out gateway.crt -days 365 -CAcreateserial -sha256 -days 365
+```
+
+5. In the S3 bucket generated in Step 1, add the `root-ca.crt` in either the `common` folder or the target tenant folder where your gateway is registered. 
+
+Now restart the Gateway Server, Identity Server and Gateway Configuration Servers.
+
+6. Connect using a client (ex: `cURL`)
```
Comment (nejraselimovic) [inline doc/content/getting-started/aws/ecs/mutual-tls/_index.md:154]:
```suggestion
6. Connect using a client (example: `cURL`)
```

**Review by nejraselimovic — APPROVED** [review]

**Comment by KrishnaIyer** [comment]
@nejraselimovic: This shouldn't have been merged. The feature which this documentation is for is not merged yet. So the `new-in-version` item is incorrect.

**Comment by nejraselimovic** [comment]
> @nejraselimovic: This shouldn't have been merged. The feature which this documentation is for is not merged yet. So the `new-in-version` item is incorrect.

Sorry, I somehow overlooked it 🤦‍♀️ do you want me to revert the merge, or just remove the `new-in-version` tag, or add a note that the feature will soon be available?

**Comment by KrishnaIyer** [comment]
I'd prefer if you could revert this for now.

**Pull request merged**